Does not build with pipewire 1.4.0

FAILED: CMakeFiles/tg_owt.dir/src/modules/video_capture/linux/pipewire_session.cc.o 
/usr/bin/x86_64-pc-linux-gnu-g++ -DABSL_ALLOCATOR_NOTHROW=1 -DBWE_TEST_LOGGING_COMPILE_TIME_ENABLE=0 -DHAVE_NETINET_IN_H -DHAVE_SCTP -DHAVE_WEBRTC_VIDEO -DNDEBUG -DNO_MAIN_THREAD_WRAPPING -DRTC_DISABLE_TRACE_EVENTS -DRTC_ENABLE_H265 -DRTC_ENABLE_VP9 -DWEBRTC_APM_DEBUG_DUMP=0 -DWEBRTC_DUMMY_AUDIO_BUILD -DWEBRTC_ENABLE_PROTOBUF=0 -DWEBRTC_HAVE_DCSCTP -DWEBRTC_HAVE_SCTP -DWEBRTC_INCLUDE_INTERNAL_AUDIO_DEVICE -DWEBRTC_LIBRARY_IMPL -DWEBRTC_LINUX -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=1 -DWEBRTC_OPUS_SUPPORT_120MS_PTIME=1 -DWEBRTC_OPUS_VARIABLE_COMPLEXITY=0 -DWEBRTC_POSIX -DWEBRTC_USE_BUILTIN_ISAC_FLOAT -DWEBRTC_USE_H264 -DWEBRTC_USE_PIPEWIRE -Dtg_owt_EXPORTS -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e_build/gen -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/third_party/pffft/src -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/third_party/libyuv/include -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/third_party/libsrtp/include -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/third_party/libsrtp/crypto/include -I/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/third_party/libsrtp/../libsrtp_config -isystem /usr/include/gio-unix-2.0 -isystem /usr/lib64/libffi/include -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/pipewire-0.3 -isystem /usr/include/spa-0.2 -isystem /usr/include/libdrm -isystem /usr/include/opus -isystem /usr/include/openh264  -DNDEBUG -O2 -pipe  -march=meteorlake -Wno-nonnull  -fpermissive -std=gnu++20 -fPIC -Wno-deprecated-declarations -Wno-attributes -Wno-narrowing -Wno-return-type -MD -MT CMakeFiles/tg_owt.dir/src/modules/video_capture/linux/pipewire_session.cc.o -MF CMakeFiles/tg_owt.dir/src/modules/video_capture/linux/pipewire_session.cc.o.d -o CMakeFiles/tg_owt.dir/src/modules/video_capture/linux/pipewire_session.cc.o -c /var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc
/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc: In constructor ‘webrtc::videocapturemodule::PipeWireNode::PipeWireNode(webrtc::videocapturemodule::PipeWireSession*, uint32_t, const spa_dict*)’:
/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc:63:24: error: cannot convert ‘pw_proxy*’ to ‘pw_node*’
   63 |   pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
      |                        ^~~~~~
      |                        |
      |                        pw_proxy*
In file included from /usr/include/pipewire-0.3/pipewire/pipewire.h:30,
                 from /var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.h:15,
                 from /var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc:11:
/usr/include/pipewire-0.3/pipewire/node.h:189:59: note:   initializing argument 1 of ‘int pw_node_add_listener(pw_node*, spa_hook*, const pw_node_events*, void*)’
  189 | PW_API_NODE_IMPL int pw_node_add_listener(struct pw_node *object,
      |                                           ~~~~~~~~~~~~~~~~^~~~~~
In file included from /usr/include/pipewire-0.3/pipewire/core.h:77,
                 from /var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.h:14:
/usr/include/pipewire-0.3/pipewire/context.h:181:16: note: class type ‘pw_proxy’ is incomplete
  181 |         struct pw_proxy * (*func) (struct pw_core *core,
      |                ^~~~~~~~
/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc: In static member function ‘static void webrtc::videocapturemodule::PipeWireNode::OnNodeInfo(void*, const pw_node_info*)’:
/var/tmp/portage/media-libs/tg_owt-0_pre20241202/work/tg_owt-be39b8c8d0db1f377118f813f0c4bd331d341d5e/src/modules/video_capture/linux/pipewire_session.cc:97:35: error: cannot convert ‘pw_proxy*’ to ‘pw_node*’
   97 |         pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
      |                             ~~~~~~^~~~~~
      |                                   |
      |                                   pw_proxy*
/usr/include/pipewire-0.3/pipewire/node.h:208:58: note:   initializing argument 1 of ‘int pw_node_enum_params(pw_node*, int, uint32_t, uint32_t, uint32_t, const spa_pod*)’
  208 | PW_API_NODE_IMPL int pw_node_enum_params(struct pw_node *object,
      |                                          ~~~~~~~~~~~~~~~~^~~~~~
/usr/include/pipewire-0.3/pipewire/context.h:181:16: note: class type ‘pw_proxy’ is incomplete
  181 |         struct pw_proxy * (*func) (struct pw_core *core,
      |                ^~~~~~~~

--- a/src/modules/video_capture/linux/pipewire_session.cc
+++ b/src/modules/video_capture/linux/pipewire_session.cc
@@ -60,7 +60,7 @@ PipeWireNode::PipeWireNode(PipeWireSession* session,
       .param = OnNodeParam,
   };
 
-  pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
+  pw_node_add_listener(reinterpret_cast<struct pw_node*>(proxy_), &node_listener_, &node_events, this);
 }
 
 PipeWireNode::~PipeWireNode() {
@@ -94,7 +94,7 @@ void PipeWireNode::OnNodeInfo(void* data, const pw_node_info* info) {
       uint32_t id = info->params[i].id;
       if (id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
+        pw_node_enum_params(reinterpret_cast<struct pw_node*>(that->proxy_), 0, id, 0, UINT32_MAX, nullptr);
         break;
       }
     }

